### PR TITLE
feat: Make empty favorite fragment similar to Eventbrite.

### LINKED ISSUE #1430: Make empty fragment_favorite similar to Eventbrite
**Describe the feature you'd like**

Make empty fragment_favorite similar to Eventbrite with relevant options as in Eventbrite

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by angmas1**
Fixes #1430 

Changes: Made UI simillar to Eventbrite with the search option as in Eventbrite.

Screenshots for the change:
image

### CODE REVIEW

**Comment by angmas1** [comment]
@liveHarshit @nikit19 please review

File: app/src/main/java/org/fossasia/openevent/general/search/SearchViewModel.kt [hunk 1]
```diff
@@ -175,6 +175,7 @@ class SearchViewModel(
             }.subscribe({
                 mutableEvents.value = it
             }, {
+                mutableEvents.value = emptyList()
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchViewModel.kt]: Why?

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchViewModel.kt]: temporary fix for #1432

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchViewModel.kt]: If I don't do this, events list won't be initialized to empty favorite fragment won't get loaded.

File: app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteFragment.kt [hunk 2]
```diff
@@ -42,6 +50,7 @@ class FavoriteFragment : Fragment() {
     private val favoriteEventsRecyclerAdapter: FavoriteEventsRecyclerAdapter by inject(
         scope = getOrCreateScope(Scopes.FAVORITE_FRAGMENT.toString())
     )
+    val preference = Preference()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteFragment.kt]: Inject

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteFragment.kt]: resolved

**Pull request merged**